✨ Index maintainers

aronhoyer opened the pull request:
This PR adds asset maintainers to our index.

As of opening this PR, the performance is absolutely atrocious (>30s response time for a full index write), but given that it's going to run as a cron job, I don't know how much we want to bother optimizing at this point.

aronhoyer committed:
:recycle: DRY code 🤓

aronhoyer committed:
:sparkles: Index maintainers

aronhoyer committed:
:pencil2: Use correct role name for data product owner

nilsml reviewed (changes requested):
Code looks good and makes sense to me.

One concern is the way NetErrors are made on multiple levels. Make sure to test and verify that we don't get NetErrors wrapped in NetErrors.

Also make sure to test that the code works without regressions before merging in.

nilsml commented on IndexCollibraDataProducts/index.ts:
```diff
@@ -33,10 +34,15 @@ const httpTrigger: AzureFunction = async function (context: Context): Promise<vo
+    ),
     TE.chainW((data) => pipe(updateAlgolia(), E.ap(E.of(data)), TE.fromEither)),
     TE.flatten,
     TE.mapLeft(toNetError(HttpStatusCode.InternalServerError)),
```
What happens if we already have a NetError from the Get function f.ex.?

aronhoyer committed:
:bug: Prevent direct forwarding of status codes

Forwarding the status code from an HTTP request committed internally to the response from the endpoint, doesn't necessarily always make sense.

nilsml reviewed (approved):
Good work!
